op-program/client/l2/engineapi/precompiles.go

```diff
@@ -66,6 +73,75 @@ func CreatePrecompileOverrides(precompileOracle PrecompileOracle) vm.PrecompileO
 			return &precompile
 		case kzgPointEvaluationPrecompileAddress:
 			return &kzgPointEvaluationOracle{Orig: orig, Oracle: precompileOracle}
+		case blsG1AddPrecompileAddress:
+			// no size limit - fixed input
+			return &blsOperationOracle{
+				Orig:              orig,
+				Oracle:            precompileOracle,
+				checkInputSize:    checkInputExactSize(256),
+				checkOutput:       checkOutputExactSize(128),
+				precompileAddress: blsG1AddPrecompileAddress,
+			}
+		case blsG1MSMPrecompileAddress:
+			return &blsOperationOracleWithSizeLimit{
+				sizeLimit: 513760,
+				blsOperationOracle: blsOperationOracle{
+					Orig:              orig,
+					Oracle:            precompileOracle,
+					checkInputSize:    checkInputSizeMultipleOf(160),
```

Comment:
The Prague precompile doesn't admit a zero-size input and this check doesn't cover that. It wouldn't be unsound to admit an empty input here since the L1 precompile will still catch it. But it's confusing to have a partially implemented input size check.

Reply:
8061a3e